flocker/node/script.py

```diff
@@ -320,18 +302,84 @@ def flocker_dataset_agent_main():
         # would be harder to implement and harder to use.
         compute_instance_id=bytes(getpid()).decode('utf-8'),
     )
-    deployer_factory = partial(
+    return partial(
         BlockDeviceDeployer,
         block_device_api=api,
     )
-    service_factory = AgentServiceFactory(
-        deployer_factory=deployer_factory
-    ).get_service
-    agent_script = AgentScript(
-        service_factory=service_factory,
+
+
+def dataset_deployer_from_configuration(dataset_configuration, volume_service):
+    """
+    Given a dataset configuration, return a dataset deployer factory.
+
+    :param dict dataset_configuration: Desired configuration for a dataset
+        deployer.
+    :param VolumeService dataset_configuration: An already started volume
+        service.
+
+    :return: A callable which can be called with a node UUID and hostname to
+        create a dataset deployer.
+    """
+    backend_to_deployer_factory = {
+        'zfs': zfs_dataset_deployer,
+        'loopback': loopback_dataset_deployer,
+    }
+    backend = dataset_configuration['backend']
+    deployer_factory = backend_to_deployer_factory[backend]
+    return deployer_factory(
```

Comment:
I think it's important you pass through the configuration (or a sub-key of the configuration); otherwise this doesn't really help backends that need further configuration.